listings/applications/nft_auction/src/nft_auction.cairo

```diff
@@ -0,0 +1,130 @@
+    fn get_name(self: @TContractState) -> felt252;
+    fn get_symbol(self: @TContractState) -> felt252;
+    fn get_token_uri(self: @TContractState, token_id: u256) -> felt252;
+    fn balance_of(self: @TContractState, account: ContractAddress) -> u256;
+    fn owner_of(self: @TContractState, token_id: u256) -> ContractAddress;
+    fn get_approved(self: @TContractState, token_id: u256) -> ContractAddress;
+    fn is_approved_for_all(
+        self: @TContractState, owner: ContractAddress, operator: ContractAddress
+    ) -> bool;
+    fn approve(ref self: TContractState, to: ContractAddress, token_id: u256);
+    fn set_approval_for_all(ref self: TContractState, operator: ContractAddress, approved: bool);
+    fn transfer_from(
+        ref self: TContractState, from: ContractAddress, to: ContractAddress, token_id: u256
+    );
+    fn mint(ref self: TContractState, to: ContractAddress, token_id: u256);
+}
+
+#[starknet::interface]
+pub trait INFTAuction<TContractState> {
+    fn buy(ref self: TContractState, token_id: u256);
+    fn get_price(self: @TContractState) -> u64;
+}
+
+#[starknet::contract]
+pub mod NFTAuction {
+    use super::{IERC20Dispatcher, IERC20DispatcherTrait, IERC721Dispatcher, IERC721DispatcherTrait};
+    use starknet::{ContractAddress, get_caller_address, get_contract_address, get_block_timestamp};
+
+    #[storage]
+    struct Storage {
+        erc20_token: ContractAddress,
+        erc721_token: ContractAddress,
+        starting_price: u64,
+        seller: ContractAddress,
+        duration: u64,
+        discount_rate: u64,
+        start_at: u64,
+        expires_at: u64,
+        purchase_count: u128,
+        total_supply: u128
+    }
+
+    #[constructor]
+    fn constructor(
+        ref self: ContractState,
+        erc20_token: ContractAddress,
+        erc721_token: ContractAddress,
+        starting_price: u64,
+        seller: ContractAddress,
+        duration: u64,
+        discount_rate: u64,
+        total_supply: u128
+    ) {
+        assert(starting_price >= discount_rate * duration, 'starting price too low');
+
+        self.erc20_token.write(erc20_token);
+        self.erc721_token.write(erc721_token);
+        self.starting_price.write(starting_price);
+        self.seller.write(seller);
+        self.duration.write(duration);
+        self.discount_rate.write(discount_rate);
+        self.start_at.write(get_block_timestamp());
+        self.expires_at.write(get_block_timestamp() + duration * 1000);
+        self.total_supply.write(total_supply);
+    }
+
+    #[abi(embed_v0)]
+    impl NFTAuction of super::INFTAuction<ContractState> {
+        fn get_price(self: @ContractState) -> u64 {
+            let time_elapsed = (get_block_timestamp() - self.start_at.read())
+                / 1000; // Ignore milliseconds
+            let discount = self.discount_rate.read() * time_elapsed;
+            let price: u64 = self.starting_price.read() - discount;
+            price
```

Comment:
just return `self.starting_price.read() - discount`

Reply:
Can you rename all the `nft_auction` by `nft_dutch_auction` (files, modules, etc...)